src/include/parser/transformer.h

```diff
@@ -34,19 +35,27 @@ struct YieldVariable;
 
 class Transformer {
 public:
-    explicit Transformer(CypherParser::Ku_StatementsContext& root) : root{root} {}
+    Transformer(CypherParser::Ku_StatementsContext& root,
+        std::vector<extension::TransformerExtension*> transformerExtensions)
+        : root{root}, transformerExtensions{std::move(transformerExtensions)} {}
 
     std::vector<std::shared_ptr<Statement>> transform();
 
+    void registerTransformExtension(
```

Comment:
Let's just make everything public if this is inevitable when implementing parser extensions. These functions has a logical order. I would recommend not changing them.